src/components/menu/__internal__/submenu/submenu.component.js

```diff
@@ -291,6 +294,9 @@ const Submenu = React.forwardRef(
                   blockIndex: React.Children.toArray(children).findIndex(
                     (item) => item.type === ScrollableBlock
                   ),
+                  updateFocusIndex: setSubmenuFocusIndex,
+                  blockDoubleFocus,
```

Comment:
I think rather than passing this value in via the context you could do the below instead. That way you won't trigger a useEffect update in `MenuItem` when the `blockDoubleFocus` value changes.
```isFocused: !blockDoubleFocus ? submenuFocusIndex === index : undefined```

Reply:
Sounds good but change the `undefined` at the end to a `false` with it being a boolean, or just this instead
```js
isFocused: !blockDoubleFocus && submenuFocusIndex === index
```